conf/cvoc/readme.md

```diff
@@ -0,0 +1,209 @@
+corresponds to:
+
+    {"personName":"Myers, James","@id":"https://orcid.org/0000-0001-8462-650X","scheme":"ORCID"}
+    
+Here's the equivalent configuration for a skosmos service where the field has been configured to allow choices from either of two vocabularies:
+
+     {
+        "field-name": "skosterm",
+        "term-uri-field": "skosterm",
+        "cvoc-url": "https://skosmos.dev.finto.fi/",
+        "js-url": "/resources/js/skosmos.js",
+        "protocol": "skosmos",
+        "retrieval-uri": "https://skosmos.dev.finto.fi/rest/v1/data?uri={0}",
+        "allow-free-text": "false",
+        "languages":"en, uk, es, zh, ar, tr, lo, sk, th, pt, hu, pl, de, cs, it, fr, hi, ja, ro, fa",
+        "vocabs":{
+            "unesco": {
+                "vocabularyUri": "http://skos.um.es/unescothes/CS000",
+                "uriSpace": "http://skos.um.es/unescothes/"
+            },
+            "agrovoc" : {
+            	   "vocabularyUri": "http://aims.fao.org/vest-registry/kos/agrovoc",
+                   "uriSpace": "http://aims.fao.org/aos/agrovoc/"
+               }
+            }
+        },
+        "managed-fields": {},
+        "retrieval-filtering": {
+            "@context": {
+                "termName": "https://schema.org/name",
+                "vocabularyName": "https://dataverse.org/schema/vocabularyName",
+                "vocabularyUri": "https://dataverse.org/schema/vocabularyUri",
+                "lang": "@language",
+                "value": "@value"
+            },
+            "@id": {
+                "pattern": "{0}",
+                "params": ["@id"]
+            },
+            "termName": {
+                "pattern": "{0}",
+                "params": ["/graph/uri=@id/prefLabel"]
+            },
+            "vocabularyName": {
+                "pattern": "{0}",
+                "params": ["/graph/type=skos:ConceptScheme/prefLabel"]
+            },
+            "vocabularyUri": {
+                "pattern": "{0}",
+                "params": ["/graph/type=skos:ConceptScheme/uri"]
+            }
+        }
+    }
+    
+The are some changes from the ORCID case:
+* vocabs now also contains a list of vocabularyURIs. The skosmos.js Javascript uses the vocabularyURIs to query the server for terms in the specified vocabulary. Further, the script will display both a vocabulary selector and a term selector when there is more than one vocabulary listed.
+* allow-free-text is set to false to not allow input of anything except for terms in the specified vocabularies. 

```

Comment:
It sounds like you can't enter random values from the GUI when "allow-free-test" is false, but what about the API?

Reply:
Yes. The entries are checked to see that the either start with the base URI (uriSpace element in the json) appropriate for the vocabulary(ies) or, if allow-free-text is true, that they are not a URI at all.